integration/spark/src/main/java/openlineage/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -73,41 +68,69 @@ public void start(SparkListenerJobStart jobStart) {
       log.info("No execution info {}", queryExecution);
       return;
     }
-    List<Dataset> outputDatasets =
+    List<OpenLineage.Dataset> outputDatasets =
         PlanUtils.applyFirst(outputDatasetSupplier, queryExecution.optimizedPlan());
-    List<Dataset> inputDatasets =
-        JavaConversions.seqAsJavaList(
-                queryExecution.optimizedPlan().collect(PlanUtils.merge(inputDatasetSupplier)))
-            .stream()
-            .flatMap(List::stream)
-            .collect(Collectors.toList());
-    LineageEvent event =
-        LineageEvent.builder()
-            .inputs(inputDatasets)
-            .outputs(outputDatasets)
+    List<OpenLineage.Dataset> inputDatasets = getInputDatasets();
+
+    OpenLineage ol = new OpenLineage(URI.create(OpenLineageClient.OPEN_LINEAGE_CLIENT_URI));
+    OpenLineage.RunEvent event =
+        ol.newRunEventBuilder()
+            .eventTime(toZonedTime(jobStart.time()))
+            .eventType("START")
+            .inputs(
+                inputDatasets.stream()
+                    .map(SparkSQLExecutionContext::convertToInput)
+                    .collect(Collectors.toList()))
+            .outputs(
+                outputDatasets == null
+                    ? null
```

Comment:
We shouldn't be passing in `null` to the list of outputs- use an empty list instead